packages/lodestar/src/chain/chain.ts

```diff
@@ -362,6 +395,30 @@ export class BeaconChain implements IBeaconChain {
     });
   }
 
+  /**
+   * Seeds the fork choice with an anchor state.
+   * This state is set as the finalized state.
+   */
+  private async initForkChoiceFromWeakSubjectivityState(anchorState: TreeBacked<BeaconState>): Promise<void> {
+    const blockHeader = anchorState.latestBlockHeader.clone() as BeaconBlockHeader;
+    blockHeader.stateRoot = anchorState.hashTreeRoot();
+    const blockRoot = this.config.types.BeaconBlockHeader.hashTreeRoot(blockHeader);
+    const epoch = computeEpochAtSlot(this.config, anchorState.slot);
+    const blockCheckpoint = {
+      root: blockRoot,
+      // TODO: is it possible to be `epoch + n` where n >= 2?
```

Comment:
The epoch of checkpoint is not always `computeEpochAtSlot(this.config, anchorState.slot)` due to this
https://github.com/ChainSafe/lodestar/blob/master/packages/lodestar-beacon-state-transition/src/epoch/justification.ts#L54

In my case, I started with state of slot 671 and if I use `computeEpochAtSlot(this.config, anchorState.slot)` I'll receive a lot of goodbye requests with reason = 2.

This change helps me got through that but not sure how can we calculate the epoch of checkpoint given a state/BeaconBlockHeader @wemeetagain @mpetrunic

Reply:
What happens if you set checkpoints to:
```
{
   root: getBlockRoot(config, anchorState, epoch),
   epoch
}
```

Maybe check how others temas implemented this?

Reply:
other nodes will disconnect us with reason code 2 due to conditions in checking status message (our `shouldDisconnectOnStatus()` is more relaxed and not 100% follow the spec I think)

Reply:
I feel like we should check how other teams are doing that, we probably need to set checkpoints from state and refetch blocks between finalized and head slot via rpc

Reply:
> we probably need to set checkpoints from state 

yeah will do that, that's the below concern too ... In that case we don't have to create our self initiated checkpoint (`blockCheckpoint` in the above code snippet)

Reply:
> I feel like we should check how other teams are doing that

I don't see this in prysmatic and lighthouse

Reply:
https://github.com/sigp/lighthouse/pull/1468

prysm probably has Pr as well